src/Hpack/Defaults.hs

```diff
@@ -51,20 +51,21 @@ get url file = do
 formatStatus :: Status -> String
 formatStatus (Status code message) = show code ++ " " ++ B.unpack message
 
-ensure :: FilePath -> Defaults -> IO (Either String FilePath)
-ensure dir = \ case
+ensure :: FilePath -> FilePath -> Defaults -> IO (Either String FilePath)
+ensure userDataDir dir = \ case
   DefaultsGithub defaults -> do
     let
       url = defaultsUrl defaults
-      file = defaultsCachePath dir defaults
+      file = defaultsCachePath userDataDir defaults
     ensureFile file url >>= \ case
       Found -> return (Right file)
       NotFound -> return (Left $ notFound url)
       Failed err -> return (Left err)
-  DefaultsLocal (Local file) ->
-    doesFileExist file >>= \ case
-      True -> return (Right file)
-      False -> return (Left $ notFound file)
+  DefaultsLocal (Local file) -> do
+    let file' = dir </> file
```

Comment:
I avoid using `'` in identifiers because admittedly, I use vim.  They don't work well with vim's `*`-search, and while vim allows you to adjust the identifier characters, adding `'` in turn won't play nice with `c2hs` which I used a lot in the past.  For this reason I tend to use `_` instead.

But in this particular case I would even eliminate the need for it by using view patterns.  That way we make sure that we can't possibly use the wrong `file` anywhere.